### PR TITLE
New MMIcon and MMCircleIcon

### PR DESCRIPTION
**Pull request by iiLubos**
image

### CODE REVIEW

File: app/mmstyle.h [hunk 1]
```diff
@@ -178,6 +181,12 @@ class MMStyle: public QObject
      * Add new sizes here only if they need to be used in multiple files
      */
 
+    // Icon sizes
+    Q_PROPERTY( double icon16 READ icon16 CONSTANT )
+    Q_PROPERTY( double icon24 READ icon24 CONSTANT )
+    Q_PROPERTY( double icon32 READ icon32 CONSTANT )
+    Q_PROPERTY( double icon40 READ icon40 CONSTANT )
```
Comment (PeterPetrik) [inline app/mmstyle.h:188]: Based on yesteday talk with Andrej I had impression we do not have 40px icons, do you see one somewhere? 40px is not icon but image or button. I would remove it from icon sizes here and also from gallery

File: app/qml/components/MMCheckBox.qml [hunk 2]
```diff
@@ -45,6 +45,7 @@ CheckBox {
       color: control.enabled ? __style.forestColor : __style.mediumGreenColor
       visible: control.checked
       scale: control.width / (24 * __dp)
+      size: 32 * __dp
```
Comment (PeterPetrik) [inline app/qml/components/MMCheckBox.qml:48]: icon32()

File: app/qml/components/MMCheckBox.qml [hunk 3]
```diff
@@ -45,6 +45,7 @@ CheckBox {
       color: control.enabled ? __style.forestColor : __style.mediumGreenColor
       visible: control.checked
       scale: control.width / (24 * __dp)
```
Comment (PeterPetrik) [inline app/qml/components/MMCheckBox.qml:47]: what is scale?

File: app/qml/components/MMCircleIcon.qml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import QtQuick
+
+Rectangle {
+  id: root
+
+  enum Type {Edit, Checkbox, Close, SmallCheckmark}
+
+  required property int type
+  required property url source
```
Comment (PeterPetrik) [inline app/qml/components/MMCircleIcon.qml:18]: Source is set internally - this should be hidden to class API - user just select type.

File: app/qml/components/MMCircleIcon.qml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick
+
+Rectangle {
+  id: root
+
+  enum Type {Edit, Checkbox, Close, SmallCheckmark}
```
Comment (PeterPetrik) [inline app/qml/components/MMCircleIcon.qml:15]: SmallCheckmark -> Checkmark

File: app/qml/components/MMCircleIcon.qml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  required property url source
+  required property real size
+  property color iconColor: __style.forestColor
+
+  color: __style.grassColor
+  width: size
+  height: size
+  radius: size / 2
+
+  MMIcon {
```
Comment (PeterPetrik) [inline app/qml/components/MMCircleIcon.qml:27]: Do not use MMIcon here - use Image{} directly

File: app/qml/components/MMIcon.qml [hunk 7]
```diff
@@ -13,24 +13,24 @@ import Qt5Compat.GraphicalEffects
 Item {
```
Comment (PeterPetrik) [inline app/qml/components/MMIcon.qml:13]: Add here comment how to use it : never set width + height , only size.

File: app/qml/components/MMCircleIcon.qml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick
+
+Rectangle {
```
Comment (PeterPetrik) [inline app/qml/components/MMCircleIcon.qml:12]: Add here comment how to use it : never set width + height , only size.

File: gallery/qml/IconLine.qml [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import QtQuick
```
Comment (PeterPetrik) [inline gallery/qml/IconLine.qml:1]: copyright

File: gallery/qml/IconLine.qml [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    Rectangle { width: 50 * __dp; height: 50 * __dp
+      MMIcon { size: __style.icon32; source: root.source; anchors.verticalCenter: parent.verticalCenter; color: __style.forestColor
+        Rectangle { anchors.fill: parent; color: "lightGray"; z: -1; visible: root.showRect }
+      }
+    }
+    Rectangle { width: 50 * __dp; height: 50 * __dp
+      MMIcon { size: __style.icon40; source: root.source; anchors.verticalCenter: parent.verticalCenter; color: __style.forestColor
+        Rectangle { anchors.fill: parent; color: "lightGray"; z: -1; visible: root.showRect }
+      }
```
Comment (PeterPetrik) [inline gallery/qml/IconLine.qml:39]: remove 40px is not usuall for icon

File: gallery/qml/IconLine.qml [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  required property string name
+  required property url source
+  required property bool showRect
+
+  Row {
+    Rectangle { width: 100 * __dp; height: 50 * __dp
+      Image { source: root.source; anchors.verticalCenter: parent.verticalCenter
+        Rectangle { anchors.fill: parent; color: "#ff8888"; z: -1 }
+      }
+    }
```
Comment (PeterPetrik) [inline gallery/qml/IconLine.qml:20]: do not understand what this pink box is? I would remove it and add there instead one white variant of icon

File: gallery/qml/pages/IconsPage.qml [hunk 12]
```diff
@@ -33,389 +33,116 @@ ScrollView {
...
-        color: "white"
-        border.color: "gray"
-      }
-      label: Text {
-        color: "black"
-        text: parent.title
-        padding: 5
-      }
+      Column {
+        MMCheckBox { id: checkbox; text: "Show in rectangle" }
```
Comment (PeterPetrik) [inline gallery/qml/pages/IconsPage.qml:37]: what this show in rectangle does?

**Review by PeterPetrik — COMMENTED** [review]
- can you please also fix the usage of icons in the app?
- can you rebase from master?

File: gallery/qml/pages/IconsPage.qml (hunk 12 above)
Comment (iiLubos) [reply gallery/qml/pages/IconsPage.qml]: just to know the icons' margins

File: app/qml/components/MMCircleIcon.qml (hunk 6 above)
Comment (iiLubos) [reply app/qml/components/MMCircleIcon.qml:26]: better to use MMIcon due to the coloring icon in the middle, I think

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**